lib/dialects/postgres/query.js

```diff
@@ -309,6 +343,44 @@ module.exports = (function() {
         }
 
         break;
+      case '23P01':
+        // there are multiple different formats of error messages for this error code
+        // this regex should check at least two
+        match = errDetail.match(/Key \((.*?)\)=\((.*?)\)/);
+
+        if (match) {
+          fields = Utils._.zipObject(match[1].split(', '), match[2].split(', '));
+          errors = [];
+          message = 'Validation error';
+
+          Utils._.forOwn(fields, function(value, field) {
+            errors.push(new sequelizeErrors.ValidationErrorItem(
+              field + ' must be unique', 'unique violation', field, value));
+          });
+
+          if (this.callee && this.callee.__options && this.callee.__options.uniqueKeys) {
```

Comment:
Yup, this is the first time I went into sequelize internals and adding an exclusion constraint error was the first thing I've done. I will sort it out soon. :)